### PR TITLE
tepl 6.00.0

### PR DESCRIPTION
**Pull request by cho-m**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by cho-m** [comment]
Note, this is dependency for `gedit` (#77970) but the repo is now archived: https://gitlab.gnome.org/Archive/tepl

**Comment by carlocab** [comment]
https://github.com/Homebrew/homebrew-core/blob/4682fdf8ae56b437407a030b432607aefd45290b/Formula/tepl.rb#L107

Probably needs to be `-ltepl-6`.

**Comment by carlocab** [comment]
```
Error: 4 failed steps!
brew linkage --test gedit
brew test --retry --verbose gedit
brew linkage --test gnome-latex
brew test --retry --verbose gnome-latex
```
You can probably include your `gedit` version bump here, as it needs a revision bump anyway.

**Comment by cho-m** [comment]
#### Some extra info based on other resources:

`tepl` seems to have been archived/deprecated with the exact release state of v6 a bit questionable; however, other actual releases like `gedit` (https://gitlab.gnome.org/GNOME/gedit/-/blob/master/NEWS) do use the last version.

Looking at decisions made by some Linux distros:
- Fedora 34 included `tepl 6`, but they decided to fully retire the package and remove all uses of it, which involved patching `gedit`/etc (https://src.fedoraproject.org/rpms/gedit/c/8885160cf6f498c6d5bc03fc858ba1d0cb82d1eb?branch=rawhide) to remove all dependencies.
- ArchLinux has updated to `tepl 6` (https://archlinux.org/packages/extra/x86_64/tepl/)
- Gentoo also updated to `tepl 6` (https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=93460744bbc9df2ab80d8bf45a5a26431057b80c) but still maintain `tepl 5`.

Since other GNOME packages and Linux Distros are using `tepl 6`, I assume it should be fine to try to update. The state of dependencies may impact whether this is possible. These include
- `gedit 40` - still active, needs `tepl 6` but has been working on **Deteplification**.
- `gnome-latex` - archived/deprecated as well, but without another release

Either way, both `tepl` and `gnome-latex` need to be marked deprecated in Homebrew.
`gedit` is the only maintained-package update, but if any issues in CI runs, we can wait for release without `tepl` dependency.

File: Formula/gnome-latex.rb [hunk 1]
```diff
@@ -24,7 +31,14 @@ class GnomeLatex < Formula
...
   depends_on "tepl"
 
+  # Add commit to port to Tepl 6
+  patch do
+    url "https://gitlab.gnome.org/Archive/gnome-latex/-/commit/e1b01186f8a4e5d3fee4c9ccfbedd6d098517df9.diff"
+    sha256 "0d54059732cb3092f52bfb8bca6ebad24a08b86036baafb31e06aca2415517ca"
+  end
+
   def install
+    system "autoreconf", "-fvi"
```
Comment (cho-m) [inline Formula/gnome-latex.rb:41]: `gnome-latex` was archived/deprecated without another release.

I currently added the unreleased commit for `tepl 6` to source tar.
The problem is that `configure` needs to be regenerated.
The official way is to use `./autogen.sh`: https://gitlab.gnome.org/Archive/gnome-latex/-/blob/master/autogen.sh
However, that is only available in the Git repo. It appears to do:
```sh
aclocal --install || exit 1
gtkdocize --copy || exit 1
intltoolize --force --copy --automake || exit 1
autoreconf --verbose --force --install || exit 1
```
Not too sure if the other 3 are necessary when rebuilding existing `configure`.
For now, trying build with just `autoreconf` step.

As reference, ArchLinux has chosen to use Git repo directly pointing to final commit in master branch.
https://github.com/archlinux/svntogit-community/blob/packages/gnome-latex/trunk/PKGBUILD#L17

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**